File: bin/hello.py
#!/usr/bin/env python3
import click

@click.command()
@click.option(
    "--greeting",
    type=str,
    required=True,
    help="A greeting.",
)
def hello(greeting: str):

    out = open("output.txt", "w") 
    out.write(f"{greeting}, World!") 
    out.close()

if __name__ == "__main__":
    hello()

File: bin/hello.r
library(optparse)

option_list <- list(
  make_option(c("--greeting"), type = "character", help = "A greeting.", metavar = "greeting")
)

opt_parser <- OptionParser(option_list = option_list)
opts <- parse_args(opt_parser)

hello <- function(greeting) {
  out <- file("output.txt", "w")
  writeLines(paste(greeting, ", World!", sep = ""), out)
  close(out)
}

if (!is.null(opts$greeting)) {
  hello(opts$greeting)
}
